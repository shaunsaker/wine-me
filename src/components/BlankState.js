import React from "react";
import { View, Text, StyleSheet } from "react-native";
import PropTypes from "prop-types";

import styleConstants from "../assets/styleConstants";

export default function BlankState(props) {
    /*
    static get propTypes() {
        return {
            headerText: PropTypes.string,
            text: PropTypes.string,
        };
    }
*/

    return (
        <View style={styles.blankStateContainer}>
            <Text style={styles.blankStateHeaderText}>{props.headerText}</Text>
            <Text style={styles.blankStateText}>{props.text}</Text>
        </View>
    );
}

const styles = StyleSheet.create({
    blankStateContainer: {
        justifyContent: "center",
        paddingVertical: 32,
        paddingHorizontal: 16,
    },
    blankStateHeaderText: {
        fontSize: styleConstants.largeFont,
        color: styleConstants.secondary,
        ...styleConstants.secondaryFont,
        marginBottom: 16,
    },
    blankStateText: {
        fontSize: styleConstants.regularFont,
        color: styleConstants.primaryText,
        ...styleConstants.primaryFont,
    },
});
