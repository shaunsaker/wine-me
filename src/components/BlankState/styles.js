import { StyleSheet } from 'react-native';

import styleConstants from '../../styleConstants';

const styles = StyleSheet.create({
  container: { flex: 1, justifyContent: 'center', alignItems: 'center' },
  icon: {
    fontSize: 64,
    color: styleConstants.colors.secondary,
    marginBottom: styleConstants.dimensions.spacing.vertical,
  },
  titleText: {
    ...styleConstants.fonts.types.title,
    marginBottom: styleConstants.dimensions.spacing.vertical,
  },
  descriptionText: { ...styleConstants.fonts.types.paragraph },
});

export default styles;
