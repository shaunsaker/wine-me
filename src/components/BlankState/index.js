import React from 'react';
import PropTypes from 'prop-types';
import { View, Text } from 'react-native';
import Icon from 'react-native-vector-icons/MaterialIcons';

import styles from './styles';

const propTypes = {
  iconName: PropTypes.string,
  title: PropTypes.string,
  description: PropTypes.string,
};

const defaultProps = {};

const BlankState = ({ iconName, title, description }) => {
  return (
    <View style={styles.container}>
      <Icon name={iconName} style={styles.icon} />

      <Text style={styles.titleText}>{title}</Text>

      <Text style={styles.descriptionText}>{description}</Text>
    </View>
  );
};

BlankState.propTypes = propTypes;
BlankState.defaultProps = defaultProps;

export default BlankState;
