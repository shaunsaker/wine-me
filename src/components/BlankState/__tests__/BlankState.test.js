import React from 'react';
import renderer from 'react-test-renderer';

import BlankState from '..';

describe('BlankState', () => {
  const iconName = 'check';
  const title = 'Test';
  const description = 'Some more';

  describe('renders', () => {
    it('renders', () => {
      const component = renderer.create(
        <BlankState iconName={iconName} title={title} description={description} />,
      );

      expect(component).toMatchSnapshot();
    });
  });
});
