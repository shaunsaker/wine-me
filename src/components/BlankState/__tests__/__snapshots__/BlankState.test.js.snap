// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlankState renders renders 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "center",
      "flex": 1,
      "justifyContent": "center",
      "maxWidth": 300,
      "paddingVertical": 100,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": undefined,
          "fontSize": 12,
        },
        Object {
          "color": "#3D84A8",
          "fontSize": 64,
          "marginBottom": 10,
        },
        Object {
          "fontFamily": "Material Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    î—Š
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#212121",
        "fontFamily": undefined,
        "fontSize": 24,
        "fontWeight": "600",
        "marginBottom": 10,
        "textAlign": "center",
      }
    }
  >
    Test
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#212121",
        "fontSize": 17,
        "textAlign": "center",
      }
    }
  >
    Some more
  </Text>
</View>
`;
